tests/Doctrine/Annotation/TokenTest.php

```
@@ -0,0 +1,153 @@
+ */
+final class TokenTest extends TestCase
+{
+    public function testDefaults()
+    {
+        $token = new Token();
+
+        $this->assertSame(DocLexer::T_NONE, $token->getType());
+        $this->assertSame('', $token->getContent());
+    }
+
+    public function testConstructorSetsValues()
+    {
+        $type = 42;
+        $content = 'questionable';
+
+        $token = new Token(
+            $type,
+            $content
+        );
+
+        $this->assertSame($type, $token->getType());
+        $this->assertSame($content, $token->getContent());
+    }
+
+    public function testCanModifyType()
+    {
+        $type = 42;
+
+        $token = new Token();
+
+        $token->setType($type);
+
+        $this->assertSame($type, $token->getType());
+    }
+
+    /**
+     * @dataProvider providerIsType
+     *
+     * @param int       $type
+     * @param int|int[] $types
+     */
+    public function testIsTypeReturnsTrue($type, $types)
+    {
+        $token = new Token();
+
+        $token->setType($type);
+
+        $this->assertTrue($token->isType($types));
+    }
+
+    /**
+     * @return array
+     */
+    public function providerIsType()
+    {
+        return array(
+            'same' => array(
+                42,
+                42,
+            ),
+            'array-containing-same' => array(
+                42,
+                array(
+                    42,
+                    9001,
+                ),
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerIsNotType
+     *
+     * @param int       $type
+     * @param int|int[] $types
+     */
+    public function testIsTypeReturnsFalse($types, $type)
+    {
+        $token = new Token();
+
+        $token->setType($type);
+
+        $this->assertFalse($token->isType($types));
+    }
+
+    /**
+     * @return array
+     */
+    public function providerIsNotType()
+    {
+        return array(
+            'equal-but-not-same' => array(
```

No they don't, that's my point. What I mean is that, as the methods are supposed to only accept  `int|int[]`, how can you be sure that they must return `false` when you pass them a string? I think in such case the methods should throw an exception instead, which would require to update the test.